packages/components/src/internal/DropdownSection.tsx

```diff
@@ -1,14 +1,19 @@
-import React, { FC, useCallback, useMemo, useState } from 'react';
+import React, { FC, ReactNode, useCallback, useMemo, useState } from 'react';
+
+import classNames from 'classnames';
 
 import { MenuDivider, MenuItem, preventDocumentHandler } from './dropdowns';
 import { DisableableMenuItem, DisableableMenuItemProps } from './components/samples/DisableableMenuItem';
-import classNames from 'classnames';
 
 const SHOW_FILTER_CUTOFF = 10;
 
+function isString(label: string | ReactNode): label is string {
+    return typeof label === 'string';
+}
+
 export interface MenuSectionItem extends Omit<DisableableMenuItemProps, 'children'> {
     disabledMessage?: string;
-    text: string;
+    label: string | ReactNode;
```

Comment:
This can be `label: ReactNode`.

The type `ReactNode` already includes `string` by way of `ReactChild -> ReactText -> string`.